plugins/inputs/whois/whois.go

```diff
@@ -55,10 +60,32 @@ func (w *Whois) Init() error {
 	return nil
 }
 
-var domainRegex = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,253}[a-zA-Z0-9]\.[a-zA-Z]{2,}$`)
+var (
+	// For ASCII domains
+	asciiDomainRegex = regexp.MustCompile(`^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,63}$`)
+	// For Punycode domains
+	punycodeRegex = regexp.MustCompile(`^(xn--[a-zA-Z0-9-]{1,59}\.)+[a-zA-Z]{2,63}$`)
+)
 
 func isValidDomain(domain string) bool {
-	return domainRegex.MatchString(domain)
+	if len(domain) > maxDomainLength {
+		return false
+	}
+
+	// Handle standard ASCII domains
+	if asciiDomainRegex.MatchString(domain) {
+		return true
+	}
+
+	// Try to convert to Punycode (handles IDNs)
+	p := idna.New(idna.MapForLookup(), idna.StrictDomainName(true))
+	punycodeVersion, err := p.ToASCII(domain)
+	if err != nil {
+		return false
+	}
+
+	// Either match ASCII pattern or punycode pattern
+	return asciiDomainRegex.MatchString(punycodeVersion) || punycodeRegex.MatchString(punycodeVersion)
```

Comment:
I don't think it's needed to specifically match against puny code format, as that should be ASCII already..
```suggestion
	return asciiDomainRegex.MatchString(punycodeVersion)
```

Reply:
@Hipska While Punycode domains are indeed ASCII-compatible, they follow a specific format that the regular ASCII domain regex doesn't capture. After conversion, a non-ASCII domain like "café.com" becomes "xn--caf-dma.com", which matches `punycodeRegex` but not `asciiDomainRegex`.

The function correctly checks both patterns after conversion:
1. ASCII pattern - for domains that might already be valid ASCII
2. Punycode pattern - for domains with the "xn--" prefix format

Both checks are necessary for complete validation of all possible domain formats.

Reply:
Wait, why wouldn’t it match the ascii regex?

Reply:
You're right! `punycodeRegex` would only be necessary if there were a specific need to distinguish Punycode domains from regular ASCII domains. It's not needed for determining validity.